modules/mobile/app/workers/mobile/v0/vaccines_updater_job.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    # This job is run daily and pulls data from the CDC to create vaccine records
+    class VaccinesUpdaterJob
+      include Sidekiq::Worker
+
+      GROUP_NAME_URL = 'https://www2.cdc.gov/vaccines/iis/iisstandards/XML.asp?rpt=vax2vg'
+      MANUFACTURER_URL = 'https://www2a.cdc.gov/vaccines/iis/iisstandards/XML.asp?rpt=tradename'
+
+      class VaccinesUpdaterError < StandardError; end
+
+      # fetches group name and manufacturer data from the CDC and stores them in the vaccines table
+      def perform
+        logger.info('BEGIN --- Updating vaccine records from CDC')
+        results = { created: 0, updated: 0, persisted: 0 }
+
+        group_name_xml.root.children.each do |node|
+          result = process_vaccine(node)
+          results[result] += 1
+        end
+
+        if (results[:created] + results[:updated] + results[:persisted]).zero?
+          raise VaccinesUpdaterError, 'No records processed'
```

Comment:
This is a canary in the coalmine error. I'm not sure that it's even possible. I tried testing the case where there are no child nodes but I wasn't able to make that happen because Nokogiri seems to add an empty child, which results in reaching the "malformed xml" error instead. But my concern is that this is a background job and there's always risk of background jobs failing silently and us not realizing it for months. I'm open to removing this, but that was my thought process in adding it.